internal/controller/controllers/clusterdeployments_controller_test.go

```diff
@@ -216,6 +227,50 @@ var _ = Describe("cluster reconcile", func() {
 				validateCreation(cluster)
 			})
 
+			It("create cluster and validate installEnv updates", func() {
+				imageInfo := models.ImageInfo{
+					DownloadURL: "downloadurl",
+				}
+				updateReply := &common.Cluster{
+					Cluster: models.Cluster{
+						ID:         clusterReply.ID,
+						Status:     swag.String(models.ClusterStatusInsufficient),
+						StatusInfo: swag.String(models.ClusterStatusInsufficient),
+					},
+					PullSecret: testPullSecretVal,
+				}
+
+				// Create Cluster
+				mockInstallerInternal.EXPECT().RegisterClusterInternal(gomock.Any(), gomock.Any(), gomock.Any()).Return(clusterReply, nil)
+				mockInstallerInternal.EXPECT().GetClusterByKubeKey(gomock.Any()).Return(clusterReply, nil).AnyTimes()
+				clusterDeployment := newClusterDeployment(clusterName, testNamespace, defaultClusterSpec)
+				Expect(c.Create(ctx, clusterDeployment)).ShouldNot(HaveOccurred())
+				validateCreation(clusterDeployment)
+				//Create InstallEnv
+				mockCRDEventsHandler.EXPECT().NotifyClusterDeploymentUpdates(gomock.Any(), gomock.Any()).Times(1)
+				mockInstallerInternal.EXPECT().GenerateClusterISOInternal(gomock.Any(), gomock.Any()).
+					Do(func(ctx context.Context, params installer.GenerateClusterISOParams) {
+						Expect(params.ClusterID).To(Equal(*clusterReply.ID))
+					}).Return(&common.Cluster{Cluster: models.Cluster{ImageInfo: &imageInfo}}, nil).Times(1)
+				installEnvImage := newInstallEnvImage("installEnvImage", testNamespace, adiiov1alpha1.InstallEnvSpec{
+					ClusterRef: &adiiov1alpha1.ClusterReference{Name: clusterDeployment.Name, Namespace: clusterDeployment.Namespace},
+					Proxy: &adiiov1alpha1.Proxy{ // These changes are expected to trigger a notification to clusterDeployments controller to Reconcile.
+						NoProxy:    "http://192.168.1.1",
+						HTTPProxy:  "http://192.168.1.2",
+						HTTPSProxy: "http://192.168.1.3",
+					},
+				})
+				Expect(c.Create(ctx, installEnvImage)).To(BeNil())
+				res, err := ir.Reconcile(newInstallEnvRequest(installEnvImage))
+				Expect(err).To(BeNil())
+				Expect(res).To(Equal(ctrl.Result{}))
+				// Reconcile clusterDepyloyment to get the updates
```

Comment:
nit: typo in comment `clusterDepyloyment`

Reply:
Done.